src/frequenz/client/dispatch/_client.py

```diff
@@ -0,0 +1,234 @@
+
+        Args:
+            grpc_channel: gRPC channel to use for communication with the API.
+            svc_addr: Address of the service to connect to.
+        """
+        self._svc_addr = svc_addr
+        self._stub = dispatch_pb2_grpc.MicrogridDispatchServiceStub(grpc_channel)
+
+    # pylint: disable=too-many-arguments, too-many-locals
+    async def list(
+        self,
+        microgrid_id: int,
+        component_selectors: list[ComponentSelector] | None = None,
+        start_from: datetime | None = None,
+        start_to: datetime | None = None,
+        end_from: datetime | None = None,
+        end_to: datetime | None = None,
+        is_active: bool | None = None,
+        is_dry_run: bool | None = None,
+    ) -> list[Dispatch]:
+        """List dispatches.
+
+        Args:
+            microgrid_id: The microgrid_id to list dispatches for.
+            component_selectors: optional, list of component ids or categories to filter by.
+            start_from: optional, filter by start_time >= start_from.
+            start_to: optional, filter by start_time < start_to.
+            end_from: optional, filter by end_time >= end_from.
+            end_to: optional, filter by end_time < end_to.
+            is_active: optional, filter by is_active status.
+            is_dry_run: optional, filter by is_dry_run status.
+
+        Returns:
+            DispatchList: List of dispatches.
+        """
+        time_interval = None
+
+        def to_timestamp(dt: datetime | None) -> Timestamp | None:
+            if dt is None:
+                return None
+
+            ts = Timestamp()
+            ts.FromDatetime(dt)
+            return ts
+
+        if start_from or start_to or end_from or end_to:
+            time_interval = PBTimeIntervalFilter(
+                start_from=to_timestamp(start_from),
+                start_to=to_timestamp(start_to),
+                end_from=to_timestamp(end_from),
+                end_to=to_timestamp(end_to),
+            )
+
+        selectors = []
+
+        for selector in component_selectors if component_selectors else []:
+            selectors.append(_component_selector_to_protobuf(selector))
+
+        filters = DispatchFilter(
+            selectors=selectors,
+            time_interval=time_interval,
+            is_active=is_active,
+            is_dry_run=is_dry_run,
+        )
+        request = DispatchListRequest(microgrid_id=microgrid_id, filter=filters)
+
+        response: Awaitable[DispatchList]
+        response = self._stub.ListMicrogridDispatches(request)  # type: ignore
+        return list(map(Dispatch.from_protobuf, (await response).dispatches))
+
+    async def create(
+        self,
+        microgrid_id: int,
+        _type: str,
+        start_time: datetime,
+        duration: timedelta,
+        selector: ComponentSelector,
+        is_active: bool,
+        is_dry_run: bool,
+        payload: dict[str, Any],
+        recurrence: RecurrenceRule,
```

Comment:
```suggestion
        recurrence: RecurrenceRule | None = None,
```
?

In this case I'm pretty sure recurrence rules are optional, but we should also document it in the protobuf spec.